lib/annotate/annotate_models.rb

```diff
@@ -515,7 +515,7 @@ def annotate_one_file(file_name, info_block, position, options = {})
       old_header = old_content.match(header_pattern).to_s
       new_header = info_block.match(header_pattern).to_s
 
-      column_pattern = /^#[\t ]+[\w\*`]+[\t ]+.+$/
+      column_pattern = /^#[\t ]+[\w\*\.`]+[\t ]+.+$/
```

Comment:
Could you leave a note as to why this would fix the problem?

Reply:
@ctran Line with a foreign key info in an annotation text looks like:

#  fk_rails_...  (foreign_thing_id => foreign_things.id) ON DELETE => restrict

It didn't match the previous pattern because of dots at the end of the foreign key name (we have dots there because we truncate generated foreign key names at https://github.com/ctran/annotate_models/blob/develop/lib/annotate/annotate_models.rb#L477).

In turn, because a foreign key line didn't match the pattern, we didn't check it for changes at https://github.com/ctran/annotate_models/blob/develop/lib/annotate/annotate_models.rb#L525

Hope this helps.

Reply:
Does it mean if the name doesn't end with "...", this pattern won't match anymore?

Reply:
@ctran No. ```[\w\*\.`]``` - this part of the regexp represents characters which a name can consist of, in any order. It will match both names with "..." and without them.